### PR TITLE
Re-enable doctest

### LINKED ISSUE #5678: doctest = don't-test
Running check/doctest on a clean client.  No tests run.  Think this is because it globs for `cirq/**/*.py` and was never updated when we moved packages!

### PR DESCRIPTION
**Pull request by dabacon**
Getting this PR out for review.  

There is a broken test due to a pending deprecation.

Fixes #5678

### CODE REVIEW

File: dev_tools/docs/run_doctest.py [hunk 1]
```diff
@@ -208,9 +222,15 @@ def import_file(file_path: str) -> ModuleType:
...
+    file_names = glob.glob('cirq**/cirq**/**/*.py', recursive=True)
+    assert file_names
     # Remove the engine client code.
-    file_names = [f for f in file_names if not f.startswith('cirq/google/engine/client/')]
+    excluded = [
+        'cirq-google/cirq_google/engine/client/',
+        'cirq-google/cirq_google/cloud/',
+        'cirq-google/cirq_google/api',
+    ]
+    file_names = [f for f in file_names if not any(f.startswith(x) for x in excluded)]
```
Comment (pavoljuhas) [inline dev_tools/docs/run_doctest.py:233]:
```suggestion
    excluded = [
        'cirq-google/cirq_google/engine/client/',
        'cirq-google/cirq_google/cloud/',
        'cirq-google/cirq_google/api/',
    ]
    file_names = [f for f in file_names if not any(f.startswith(x) for x in excluded)]
    file_names = [f for f in file_names if not f.endswith('_test.py')]
```

File: dev_tools/docs/run_doctest.py [hunk 2]
```diff
@@ -161,6 +174,7 @@ def exec_tests(
     failed, attempted = 0, 0
     error_messages = []
     for test in tests:
+        print(test.file_name)
```
Comment (pavoljuhas) [inline dev_tools/docs/run_doctest.py:177]: Please check if the printout should be in.

**Review by pavoljuhas — APPROVED** [review]
LGTM if CI checks pass.

File: dev_tools/docs/run_doctest.py (hunk 1 above)
Comment (dabacon) [reply dev_tools/docs/run_doctest.py:233]: Oh yeah good, idea skipping tests.  Added skipping.

File: dev_tools/docs/run_doctest.py [hunk 3]
```diff
@@ -230,17 +229,19 @@ def main():
         'cirq-google/cirq_google/cloud/',
         'cirq-google/cirq_google/api',
```
Comment (pavoljuhas) [inline dev_tools/docs/run_doctest.py:230]: To be on a safe side - let's use replace `api` --> `api/` as for above paths.

Comment (dabacon) [reply dev_tools/docs/run_doctest.py]: fixed

**Pull request merged**